File: umbrella-insurance-react-ui/src/models/users/phoneNumbersHistory/v1/PhoneNumberHistory.ts
import { User } from "../../v1/User";

export class PhoneNumberHistory {
      id?: number;
      user?: User;
      phoneNumber?: string;
      createdDateTime?: string;

      constructor(obj?:any) {
            if(obj) {
                if(obj?.id) {
                    this.id = obj.id;
                }
                if(obj?.user) {
                    this.user = new User(obj.user);
                }
                if(obj?.phoneNumber) {
                    this.phoneNumber = obj.phoneNumber;
                }
                if(obj?.createdDateTime) {
                    this.createdDateTime = obj.createdDateTime;
                }
            }
        }
}

File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/PhoneNumberHistory.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "phone_numbers_history", schema = "public")
public class PhoneNumberHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "phone_number_history_id", nullable = false)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "user_id")
    private User user;

    @NotNull
    @Column(name = "phone_number", nullable = false, length = Integer.MAX_VALUE)
    private String phoneNumber;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @Column(name = "created_date_time")
    private Timestamp createdDateTime;

}